src/extension/ddappsec.c

```diff
@@ -295,26 +316,34 @@ static PHP_RSHUTDOWN_FUNCTION(ddappsec)
     UNUSED(type);
     UNUSED(module_number);
 
-    if (!DDAPPSEC_G(enabled)) {
-        return SUCCESS;
+    ZEND_RESULT_CODE result = SUCCESS;
+
+    // Here now we have to disconnect from the helper in all the cases but when
+    // disabled by config
+    if (DDAPPSEC_G(enabled_by_configuration) == DISABLED) {
+        goto exit;
     }
 
     if (DDAPPSEC_G(skip_rshutdown)) {
-        return SUCCESS;
+        goto exit;
     }
 
     if (UNEXPECTED(get_global_DD_APPSEC_TESTING())) {
         dd_tags_rshutdown_testing();
-        return SUCCESS;
+        goto exit;
     }
 
-    return dd_appsec_rshutdown();
+    result = dd_appsec_rshutdown();
```

Comment:
I think the cleanup operations are a bit more complex than what you've done here. In the case in which request_init is called, you're leaking the contents that would normally be cleaned up with `dd_tags_rshutdown`.

Reply:
See what I said on the next comment below